[CI] PromqlParserTests testInstantVectorExpected failing

**Build Scans:**
- [elasticsearch-intake #34574 / part3](https://gradle-enterprise.elastic.co/s/tecjcymfv4d7y)
- [elasticsearch-pull-request #120738 / part-3](https://gradle-enterprise.elastic.co/s/tt6j4rbkgkzpw)
- [elasticsearch-intake #34573 / part3](https://gradle-enterprise.elastic.co/s/wbwuuxoqsipjk)
- [elasticsearch-pull-request #120730 / part-3](https://gradle-enterprise.elastic.co/s/ig7wbzuidpxba)
- [elasticsearch-pull-request #120700 / part-3](https://gradle-enterprise.elastic.co/s/jddwhxise7o7m)
- [elasticsearch-pull-request #120278 / part-3](https://gradle-enterprise.elastic.co/s/bc7nrw5ktmcda)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.parser.promql.PromqlParserTests.testInstantVectorExpected" -Dtests.seed=91E1C3C4F10C33AE -Dtests.locale=mgh-MZ -Dtests.timezone=Asia/Srednekolymsk -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.parser.promql.PromqlParserTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testInstantVectorExpected'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a string containing "expected type instant vector in call to function [avg], got range vector"
     but: was "line 1:53: expected type instant_vector in call to function [avg], got range_vector"
```

**Issue Reasons:**
- [main] 2 consecutive failures in step part3
- [main] 2 consecutive failures in pipeline elasticsearch-intake
- [main] 6 failures in test testInstantVectorExpected (2.0% fail rate in 303 executions)
- [main] 4 failures in step part-3 (1.9% fail rate in 206 executions)
- [main] 2 failures in step part3 (4.4% fail rate in 45 executions)
- [main] 4 failures in pipeline elasticsearch-pull-request (1.9% fail rate in 206 executions)
- [main] 2 failures in pipeline elasticsearch-intake (4.4% fail rate in 45 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -389,18 +389,6 @@ tests:
 - class: org.elasticsearch.benchmark.vector.scorer.VectorScorerOSQBenchmarkTests
   method: testBulkScalarVsVectorized {p0=768 p1=1 p2=MMAP p3=MAXIMUM_INNER_PRODUCT}
   issue: https://github.com/elastic/elasticsearch/issues/142413
-- class: org.elasticsearch.xpack.esql.expression.promql.function.PromqlKibanaDefinitionGeneratorTests
-  method: testGenerateDefinitions
-  issue: https://github.com/elastic/elasticsearch/issues/142417
-- class: org.elasticsearch.xpack.esql.parser.promql.PromqlParserTests
-  method: testInstantVectorExpectedWithGrouping
-  issue: https://github.com/elastic/elasticsearch/issues/142418
-- class: org.elasticsearch.xpack.esql.parser.promql.PromqlParserTests
-  method: testRangeVectorExpected
-  issue: https://github.com/elastic/elasticsearch/issues/142419
-- class: org.elasticsearch.xpack.esql.parser.promql.PromqlParserTests
-  method: testInstantVectorExpected
-  issue: https://github.com/elastic/elasticsearch/issues/142420
 - class: org.elasticsearch.indices.stats.IndexStatsIT
   method: testFieldDataStats
   issue: https://github.com/elastic/elasticsearch/issues/142424

